### PR TITLE
Work with Junit try#1

### PR DESCRIPTION
**Pull request by OsPit01**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>groupId</groupId>
```
Comment (VladOsipov) [inline pom.xml]: change group id

File: src/main/java/Main.java [hunk 2]
```diff
@@ -2,7 +2,7 @@
 import command.LoginCommand;
 import command.PrintUsersCommand;
 import command.RegisterCommand;
-import command.constant.CommandConstants;
+import command.constatnts.CommandConstants;
```
Comment (VladOsipov) [inline src/main/java/Main.java]: constant

File: src/main/java/converter/StringToUserConverter.java [hunk 3]
```diff
@@ -7,9 +7,10 @@ public class StringToUserConverter {
     public User convert(String line) {
 
         if (line == null || line.isBlank()) {
-            throw new IllegalArgumentException("string = null");
+            throw new IllegalArgumentException("wrong user created");
```
Comment (VladOsipov) [inline src/main/java/converter/StringToUserConverter.java]: previous message was better

File: src/main/java/converter/StringToUserConverter.java [hunk 4]
```diff
@@ -7,9 +7,10 @@ public class StringToUserConverter {
...
         }
-        String[] fields = line.split(ConverterConstants.USER_FIELDS_SEPARATOR);
+
+        String[] fields = line.split("=");
```
Comment (VladOsipov) [inline src/main/java/converter/StringToUserConverter.java]: use constant

File: pom.xml [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    </properties>
+    <dependencies>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-testng</artifactId>
+            <version>0.4.30</version>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (VladOsipov) [inline pom.xml]: Add empty line

File: src/test/java/converter/UserToStringConverterTest.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+*/
+    @Test
+    public void testConvert() {
+        User user = new User("igor", "igor_password", "ihar", "osipov");
+        String actual = userToStringConverter.convert(user);
+        assertNotNull(actual);
+        String expected = "igor=igor_password=ihar=osipov";
+        assertEquals(expected, actual);
+    }
+}
```
Comment (VladOsipov) [inline src/test/java/converter/UserToStringConverterTest.java]: Add empty line

File: src/test/java/converter/UserToStringConverterTest.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+
+public class UserToStringConverterTest {
+
+    final UserToStringConverter userToStringConverter = new UserToStringConverter();
+
```
Comment (VladOsipov) [inline src/test/java/converter/UserToStringConverterTest.java:24]: Remove commented code

File: src/test/java/converter/UserToStringConverterTest.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import model.User;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+
+public class UserToStringConverterTest {
+
+    final UserToStringConverter userToStringConverter = new UserToStringConverter();
```
Comment (VladOsipov) [inline src/test/java/converter/UserToStringConverterTest.java]: private/public

File: src/test/java/converter/UserToStringConverterTest.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+
+    /*  @Test
+  public void createdUser() {
+User user = new User("igor","igor_password","ihar","osipov");
+User newUser = Mockito.mock(User.class);
+assertEquals("ihar",newUser.getSurname());
+   }
+*/
+    @Test
+    public void testConvert() {
```
Comment (VladOsipov) [inline src/test/java/converter/UserToStringConverterTest.java:26]: Add test for converting null object and user with null fields

File: src/test/java/converter/UserToStringConverterTest.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+        User user = new User("igor", "igor_password", "ihar", "osipov");
+        String actual = userToStringConverter.convert(user);
```
Comment (VladOsipov) [inline src/test/java/converter/UserToStringConverterTest.java:30]: separate with empty line

File: src/test/java/converter/UserToStringConverterTest.java [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+        assertNotNull(actual);
+        String expected = "igor=igor_password=ihar=osipov";
```
Comment (VladOsipov) [inline src/test/java/converter/UserToStringConverterTest.java]: Expected should be in the first section of test

File: src/test/java/converter/UserToStringConverterTest.java [hunk 12]
```diff
@@ -0,0 +1,29 @@
+package converter;
+
+import model.User;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+
+public class UserToStringConverterTest {
```
Comment (VladOsipov) [inline src/test/java/converter/UserToStringConverterTest.java:9]: should be not public

File: src/test/java/converter/StringToUserConverterTest.java [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class StringToUserConverterTest {
+    final StringToUserConverter stringToUserConverter = new StringToUserConverter();
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertForEmptyString() {
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: testConvertEmptyString

File: src/test/java/converter/StringToUserConverterTest.java [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+        stringToUserConverter.convert(" ");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertForEmptyLines() {
+        String actual = "" + "" + "" + "";
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: “” + “” + “” + “” = “”

File: src/test/java/converter/StringToUserConverterTest.java [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertForNullString() {
+        stringToUserConverter.convert(null);
+    }
+
+    @Test
+    public void testConvertNameWithDash() {
+        String expected = "Jana-dark";
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: Expected should be a User. As convert method returns User

File: src/test/java/converter/StringToUserConverterTest.java [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+        User actual = stringToUserConverter.convert("Jana-dark=janka_password=Vany=Vasilevna");
+
+        assertNotNull(actual);
+        assertEquals("can be: Jana-dark", expected, actual.getUsername());
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: Verify allUser fields. Not only name

File: src/test/java/converter/StringToUserConverterTest.java [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    @Test
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: Redundant. Slash and underslash are the same

File: src/test/java/converter/StringToUserConverterTest.java [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+        User expected = new User("Oss", "oss_password", "igor", "osipov");
+        User actual = stringToUserConverter.convert("Oss=oss_password=igor=osipov");
+
+        assertNotNull(actual);
+        assertEquals("can be userName: Oss", expected.getUsername(), actual.getUsername());
+        assertEquals("can be password: oss_password", expected.getPassword(), actual.getPassword());
+        assertEquals("can be name: igor", expected.getName(), actual.getName());
+        assertEquals("can be name: osipov", expected.getSurname(), actual.getSurname());
+    }
+
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: 2 empty lines

File: src/test/java/converter/StringToUserConverterTest.java [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
+
+}
+
+
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: 2 empty lines

**Review by VladOsipov — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (OsPit01) [reply pom.xml]: correted

File: src/main/java/Main.java (hunk 2 above)
Comment (OsPit01) [reply src/main/java/Main.java]: corrected

File: src/main/java/converter/StringToUserConverter.java (hunk 3 above)
Comment (OsPit01) [reply src/main/java/converter/StringToUserConverter.java]: corrected

File: src/main/java/converter/StringToUserConverter.java (hunk 4 above)
Comment (OsPit01) [reply src/main/java/converter/StringToUserConverter.java]: corrected

File: src/test/java/converter/UserToStringConverterTest.java (hunk 6 above)
Comment (OsPit01) [reply src/test/java/converter/UserToStringConverterTest.java]: corrected

File: src/test/java/converter/UserToStringConverterTest.java (hunk 7 above)
Comment (OsPit01) [reply src/test/java/converter/UserToStringConverterTest.java:24]: corrected

File: src/test/java/converter/UserToStringConverterTest.java (hunk 8 above)
Comment (OsPit01) [reply src/test/java/converter/UserToStringConverterTest.java]: corrected

File: src/test/java/converter/UserToStringConverterTest.java (hunk 10 above)
Comment (OsPit01) [reply src/test/java/converter/UserToStringConverterTest.java:30]: corrected

File: src/test/java/converter/UserToStringConverterTest.java (hunk 11 above)
Comment (OsPit01) [reply src/test/java/converter/UserToStringConverterTest.java]: corrected

File: src/test/java/converter/UserToStringConverterTest.java (hunk 12 above)
Comment (OsPit01) [reply src/test/java/converter/UserToStringConverterTest.java:9]: corrected = delete public

Comment (OsPit01) [reply src/test/java/converter/UserToStringConverterTest.java:9]: test not impossible without public

File: src/test/java/converter/StringToUserConverterTest.java (hunk 13 above)
Comment (OsPit01) [reply src/test/java/converter/StringToUserConverterTest.java]: corected

File: src/test/java/converter/StringToUserConverterTest.java (hunk 15 above)
Comment (OsPit01) [reply src/test/java/converter/StringToUserConverterTest.java:30]: corrected

File: src/test/java/converter/StringToUserConverterTest.java (hunk 17 above)
Comment (OsPit01) [reply src/test/java/converter/StringToUserConverterTest.java]: corrected

File: src/test/java/converter/StringToUserConverterTest.java (hunk 18 above)
Comment (OsPit01) [reply src/test/java/converter/StringToUserConverterTest.java]: corrected

File: src/test/java/converter/StringToUserConverterTest.java (hunk 19 above)
Comment (OsPit01) [reply src/test/java/converter/StringToUserConverterTest.java]: corrected

File: src/test/java/converter/StringToUserConverterTest.java (hunk 14 above)
Comment (OsPit01) [reply src/test/java/converter/StringToUserConverterTest.java:19]: corrected

File: pom.xml (hunk 5 above)
Comment (OsPit01) [reply pom.xml]: corrected

File: src/test/java/converter/UserToStringConverterTest.java (hunk 9 above)
Comment (OsPit01) [reply src/test/java/converter/UserToStringConverterTest.java:26]: corrected

File: src/test/java/converter/StringToUserConverterTest.java (hunk 16 above)
Comment (OsPit01) [reply src/test/java/converter/StringToUserConverterTest.java]: create

File: src/main/java/converter/StringToUserConverter.java [hunk 20]
```diff
@@ -7,8 +7,9 @@ public class StringToUserConverter {
     public User convert(String line) {
 
         if (line == null || line.isBlank()) {
-            throw new IllegalArgumentException("string = null");
+            throw new IllegalArgumentException("String = null");
```
Comment (VladOsipov) [inline src/main/java/converter/StringToUserConverter.java]:
```suggestion
            throw new IllegalArgumentException("line = null");
```
as your check line in `if`, not string

File: pom.xml [hunk 21]
```diff
@@ -0,0 +1,29 @@
...
+    <version>1.0-SNAPSHOT</version>
+
+    <properties>
+        <maven.compiler.source>18</maven.compiler.source>
+        <maven.compiler.target>18</maven.compiler.target>
+    </properties>
+    <dependencies>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-testng</artifactId>
```
Comment (VladOsipov) [inline pom.xml]: you don't need this dependency now

File: pom.xml [hunk 22]
```diff
@@ -0,0 +1,29 @@
...
+            <version>0.4.30</version>
+        </dependency>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
```
Comment (VladOsipov) [inline pom.xml]: why do you need this dependency?

Also you don't have junit dependency

File: src/test/java/converter/UserToStringConverterTest.java (hunk 8 above)
Comment (VladOsipov) [reply src/test/java/converter/UserToStringConverterTest.java]: check every place

File: src/test/java/converter/StringToUserConverterTest.java [hunk 23]
```diff
@@ -0,0 +1,94 @@
...
+    final StringToUserConverter stringToUserConverter = new StringToUserConverter();
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertEmptyString() {
+        stringToUserConverter.convert(" ");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertEmptyFields() {
+        String actual = "" + " " + "" + " " + "" + "";
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: “” + “” + “” + “” = “”

File: src/test/java/converter/StringToUserConverterTest.java [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class StringToUserConverterTest {
+    final StringToUserConverter stringToUserConverter = new StringToUserConverter();
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertEmptyString() {
+        stringToUserConverter.convert(" ");
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: `""` - empty String
`" "` - blank String

File: src/test/java/converter/StringToUserConverterTest.java [hunk 25]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertNullString() {
+        stringToUserConverter.convert(null);
+    }
+
+
+    @Test
+    public void testConvertNameWithUnderSlash() {
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: `-` and `_` are the same cases
but I'd recommend you to make test for `-` as it is more likely

File: src/test/java/converter/StringToUserConverterTest.java [hunk 26]
```diff
@@ -0,0 +1,94 @@
...
+        User expected = stringToUserConverter.convert("Van_Clock_Dimon=dimon=vany=rbIba");
+
+        String actual = "Van_Clock_Dimon";
+
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: 2 empty lines

File: src/test/java/converter/StringToUserConverterTest.java [hunk 27]
```diff
@@ -0,0 +1,94 @@
...
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertNullString() {
+        stringToUserConverter.convert(null);
+    }
+
+
+    @Test
+    public void testConvertNameWithUnderSlash() {
+        User expected = stringToUserConverter.convert("Van_Clock_Dimon=dimon=vany=rbIba");
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: should be in the second block

File: src/test/java/converter/StringToUserConverterTest.java [hunk 28]
```diff
@@ -0,0 +1,94 @@
...
+
+        String actual = "Van_Clock_Dimon";
+
+
+        assertNotNull(actual);
+        assertEquals("can be:Van_Clock_Dimon", expected.getUsername(), actual);
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: Onlu username can be with underslah?
Do we need to verify other fields

File: src/test/java/converter/StringToUserConverterTest.java [hunk 29]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    @Test
+    public void testConvertNameOfThreeWords() {
+        User expected = stringToUserConverter.convert("serega=vany_password=Van Clock Vandam=pushkin");
+
+        String actual = "Van Clock Vandam";
+
+        assertNotNull(actual);
+        assertEquals("can be: Van Clock Vandam", expected.getName(), actual);
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: can surname be with more than one word?

File: src/test/java/converter/StringToUserConverterTest.java [hunk 30]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    @Test
+    public void testConvertSurnameOfThreeWords() {
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: you can do this check in one test

File: src/test/java/converter/StringToUserConverterTest.java [hunk 31]
```diff
@@ -0,0 +1,94 @@
...
+    }
+
+    @Test
+    public void testConvert() {
+        User expected = new User("Oss", "oss_password", "igor", "osipov");
+
+        User actual = stringToUserConverter.convert("Oss=oss_password=igor=osipov");
+
+        assertNotNull(actual);
+        assertEquals("can be userName: Oss", expected.getUsername(), actual.getUsername());
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]:
```suggestion
        assertEquals("expected: Oss", expected.getUsername(), actual.getUsername());
```

or remove strings at all
```suggestion
        assertEquals(expected.getUsername(), actual.getUsername());
```

File: src/main/java/Main.java [hunk 32]
```diff
@@ -2,7 +2,7 @@
 import command.LoginCommand;
 import command.PrintUsersCommand;
 import command.RegisterCommand;
-import command.constant.CommandConstants;
+import command.constatnt.CommandConstants;
```
Comment (VladOsipov) [inline src/main/java/Main.java]: constant

File: src/main/java/converter/UserToStringConverter.java [hunk 33]
```diff
@@ -7,6 +7,9 @@
 public class UserToStringConverter {
 
     public String convert(User user) {
+        if (user == null) {
+            throw new IllegalArgumentException("String = null");
```
Comment (VladOsipov) [inline src/main/java/converter/UserToStringConverter.java]: what string is null?

File: src/test/java/converter/StringToUserConverterTest.java [hunk 34]
```diff
@@ -0,0 +1,94 @@
...
+    public void testConvertNullString() {
+        stringToUserConverter.convert(null);
+    }
+
+
+    @Test
+    public void testConvertNameWithUnderSlash() {
+        User expected = stringToUserConverter.convert("Van_Clock_Dimon=dimon=vany=rbIba");
+
+        String actual = "Van_Clock_Dimon";
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: test method returns `actual`
you define `expected`

File: src/test/java/converter/UserToStringConverterTest.java (hunk 9 above)
Comment (VladOsipov) [reply src/test/java/converter/UserToStringConverterTest.java:26]: see only for null user
what string will be written in file in case `User(null, null, null, null)` ?

**Review by VladOsipov — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 35]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+
```
Comment (VladOsipov) [inline pom.xml]: 2 empty lines

File: src/main/java/converter/UserToStringConverter.java [hunk 36]
```diff
@@ -7,6 +7,12 @@
 public class UserToStringConverter {
 
     public String convert(User user) {
+        if (user == null) {
```
Comment (VladOsipov) [inline src/main/java/converter/UserToStringConverter.java]: what about moving this validations to validation method?
ex: `validate(user);`

File: src/test/java/converter/StringToUserConverterTest.java (hunk 23 above)
Comment (VladOsipov) [reply src/test/java/converter/StringToUserConverterTest.java]: still actual

File: src/test/java/converter/StringToUserConverterTest.java [hunk 37]
```diff
@@ -0,0 +1,122 @@
...
+        stringToUserConverter.convert(actual);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertNullString() {
+        stringToUserConverter.convert(null);
+    }
+
+    @Test
+    public void testConvertNameWithDash() {
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: only name? What about surname with dash?

Comment (VladOsipov) [reply src/test/java/converter/StringToUserConverterTest.java]: I see separate test for surname. Why? It is one test for dash in fields

File: src/test/java/converter/StringToUserConverterTest.java (hunk 30 above)
Comment (VladOsipov) [reply src/test/java/converter/StringToUserConverterTest.java]: still actual. the same as for dash

File: src/test/java/converter/StringToUserConverterTest.java (hunk 31 above)
Comment (VladOsipov) [reply src/test/java/converter/StringToUserConverterTest.java]: nothing changed

File: src/test/java/converter/UserToStringConverterTest.java [hunk 38]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Test
+    public void testConvert() {
+        String expected = "igor=igor_password=ihar=osipov";
+        User user = new User("igor", "igor_password", "ihar", "osipov");
+
+        String actual = userToStringConverter.convert(user);
+
+        assertNotNull(actual);
+        assertEquals(expected,actual);
```
Comment (VladOsipov) [inline src/test/java/converter/UserToStringConverterTest.java]: here you compare user with user
but in other tests you use getters. Why? Maybe we should make all tests looks the same

**Review by VladOsipov — CHANGES_REQUESTED** [review]

File: src/test/java/converter/StringToUserConverterTest.java (hunk 16 above)
Comment (VladOsipov) [reply src/test/java/converter/StringToUserConverterTest.java]: still actual

File: src/main/java/converter/StringToUserConverter.java (hunk 20 above)
Comment (VladOsipov) [reply src/main/java/converter/StringToUserConverter.java]: still actual

File: src/main/java/converter/StringToUserConverter.java [hunk 39]
```diff
@@ -6,8 +6,9 @@ public class StringToUserConverter {
 
     public User convert(String line) {
 
+
```
Comment (VladOsipov) [inline src/main/java/converter/StringToUserConverter.java]: 2 lines

File: src/main/java/converter/StringToUserConverter.java [hunk 40]
```diff
@@ -6,8 +6,9 @@ public class StringToUserConverter {
...
         if (line == null || line.isBlank()) {
-            throw new IllegalArgumentException("String = null or Blank");
+            throw new IllegalArgumentException("line = null");
```
Comment (VladOsipov) [inline src/main/java/converter/StringToUserConverter.java]: You throw an exception when line is null OR empty. 
String.format(“Illegal argument line = [%]”, line)

File: pom.xml [hunk 41]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
```
Comment (VladOsipov) [inline pom.xml:6]: 2 lines

File: src/main/java/converter/StringToUserConverter.java (hunk 40 above)
Comment (VladOsipov) [reply src/main/java/converter/StringToUserConverter.java]: And check other messages

File: src/test/java/converter/StringToUserConverterTest.java [hunk 42]
```diff
@@ -0,0 +1,71 @@
...
+    private final StringToUserConverter stringToUserConverter = new StringToUserConverter();
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertEmptyString() {
+        stringToUserConverter.convert("");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertBlankString() {
+        String actual = " ";
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: testComvertEmptyString doesn’t have actual
But this methods are the same. Should looks the same

File: src/test/java/converter/StringToUserConverterTest.java [hunk 43]
```diff
@@ -0,0 +1,71 @@
...
+        stringToUserConverter.convert(actual);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertNullString() {
+        stringToUserConverter.convert(null);
+    }
+
+    @Test
+    public void testConvertUser() {
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: ?

File: src/test/java/converter/StringToUserConverterTest.java [hunk 44]
```diff
@@ -0,0 +1,71 @@
...
+
+        assertEquals("can be User: null", expected, actual);
+    }
+
+    @Test
+    public void testConvertWithDash() {
+        User actual = stringToUserConverter.convert("vany-kaka=dimon-maradona=Van-Clock-Dimon=rbIba-gold");
+
+        assertNotNull(actual);
+        assertEquals("can be name:Van-Clock-Dimon", "Van-Clock-Dimon", actual.getName());
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: make variables for expected results

File: src/test/java/converter/UserToStringConverterTest.java [hunk 45]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Test
+    public void testConvert() {
+        String expected = "igor=igor_password=ihar=osipov";
+        User user = new User("igor", "igor_password", "ihar", "osipov");
+
+        String actual = userToStringConverter.convert(user);
+
+        assertNotNull(actual);
+        assertEquals("String can be:igor=igor_password=ihar=osipov", expected, actual);
```
Comment (VladOsipov) [inline src/test/java/converter/UserToStringConverterTest.java]: Use getters

**Review by VladOsipov — CHANGES_REQUESTED** [review]

File: src/test/java/converter/StringToUserConverterTest.java (hunk 44 above)
Comment (VladOsipov) [reply src/test/java/converter/StringToUserConverterTest.java]: User expected = ...
User actual = ...

assertEquasl(expected.getSmth(), actual.getSmth())

File: src/test/java/converter/UserToStringConverterTest.java (hunk 45 above)
Comment (VladOsipov) [reply src/test/java/converter/UserToStringConverterTest.java]: need the same style everywhere
in other places you use getters

File: src/main/java/converter/UserToStringConverter.java [hunk 46]
```diff
@@ -1,17 +1,25 @@
...
 
 import java.util.StringJoiner;
 
 public class UserToStringConverter {
+     private final UserValidator validator = new UserValidator();
+
 
     public String convert(User user) {
+        if (!(validator.isValid(user)))  {  //todo move to another laye
+            throw new IllegalArgumentException("user = null");
```
Comment (VladOsipov) [inline src/main/java/converter/UserToStringConverter.java]: invalid message. user can be not null and we will get this exception

File: src/test/java/converter/StringToUserConverterTest.java [hunk 47]
```diff
@@ -0,0 +1,71 @@
...
+    @Test
+    public void testConvertUser() {
+        User actual = null;
+        User expected = null;
+
+        assertEquals("can be User: null", expected, actual);
+    }
+
+    @Test
+    public void testConvertWithDash() {
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]:
```suggestion
    public void testConvertStringWithDash() {
```

**Review by VladOsipov — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 41 above)
Comment (VladOsipov) [inline pom.xml:6]: 2 lines

File: src/main/java/converter/StringToUserConverter.java [hunk 48]
```diff
@@ -7,8 +7,10 @@ public class StringToUserConverter {
     public User convert(String line) {
 
         if (line == null || line.isBlank()) {
-            throw new IllegalArgumentException("string = null");
+
+            throw new IllegalArgumentException(String.format("Illegal argument line = [%s]",line));
```
Comment (VladOsipov) [inline src/main/java/converter/StringToUserConverter.java]: Ctrl + alt + L

File: src/main/java/converter/UserToStringConverter.java [hunk 49]
```diff
@@ -1,12 +1,19 @@
...
 
 import java.util.StringJoiner;
 
 public class UserToStringConverter {
+    private final UserValidator validator = new UserValidator();
+
 
     public String convert(User user) {
+        if (!(validator.isValid(user))) {  //todo move to another laye
+            throw new IllegalArgumentException(String.format("Illegal argument user invalid user = [%s]", user.toString()));
```
Comment (VladOsipov) [inline src/main/java/converter/UserToStringConverter.java:14]: Check if it works without toString

Comment (VladOsipov) [inline src/main/java/converter/UserToStringConverter.java]: Choose only one:
- Illegal argument user
- Invalid user

File: src/test/java/converter/StringToUserConverterTest.java [hunk 50]
```diff
@@ -0,0 +1,57 @@
...
+public class StringToUserConverterTest {
+    private final StringToUserConverter stringToUserConverter = new StringToUserConverter();
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertEmptyString() {
+        stringToUserConverter.convert("");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertNullString() {
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java:22]: What about test of blank string?

File: src/test/java/converter/StringToUserConverterTest.java [hunk 51]
```diff
@@ -0,0 +1,57 @@
...
+        stringToUserConverter.convert(null);
+    }
+
+    @Test
+    public void  testConvertStringWithDash() {
+        User expected = new User("vany-kaka", "dimon-maradona", "Van-Clock-Dimon", "rbIba-gold");
+        User actual = stringToUserConverter.convert("vany-kaka=dimon-maradona=Van-Clock-Dimon=rbIba-gold");
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java:35]: Separate test sections with empty lines

File: src/test/java/converter/StringToUserConverterTest.java [hunk 52]
```diff
@@ -0,0 +1,57 @@
...
+    @Test
+    public void testConvert() {
+        User expected = new User("Oss", "oss_password", "igor", "osipov");
+        User actual = stringToUserConverter.convert("Oss=oss_password=igor=osipov");
+
+        assertNotNull(actual);
+        assertEquals(expected.getUsername(), actual.getUsername());
+        assertEquals(expected.getPassword(), actual.getPassword());
+        assertEquals(expected.getName(), actual.getName());
+        assertEquals(expected.getSurname(), actual.getSurname());
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java:55]: No need to separate asserts with empty lines

File: src/test/java/converter/StringToUserConverterTest.java [hunk 53]
```diff
@@ -0,0 +1,57 @@
...
+
+        assertNotNull(actual);
+        assertEquals(expected.getName(), actual.getName());
+        assertEquals(expected.getSurname(), actual.getSurname());
+        assertEquals(expected.getPassword(), actual.getPassword());
+        assertEquals(expected.getUsername(), actual.getUsername());
+    }
+
+    @Test
+    public void testConvertUserOfThreeWords() {
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: Change name. User -> String 
Convert method converts string. Not a user

**Review by VladOsipov — CHANGES_REQUESTED** [review]

File: src/test/java/converter/StringToUserConverterTest.java [hunk 54]
```diff
@@ -0,0 +1,67 @@
...
+    public void testConvertNullString() {
+        stringToUserConverter.convert(null);
+    }
+
+    @Test
+    public void testConvertStringWithDash() {
+        User expected = new User("vany-kaka",
+                "dimon-maradona",
+                "Van-Clock-Dimon",
+                "rbIba-gold");
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]:
```
                new User(
                    "vany-kaka",
                    "dimon-maradona",
                    "Van-Clock-Dimon",
                    "rbIba-gold"
                );
```

Or
```
                new User("vany-kaka", "dimon-maradona", "Van-Clock-Dimon", "rbIba-gold");
```

File: src/test/java/converter/StringToUserConverterTest.java [hunk 55]
```diff
@@ -0,0 +1,67 @@
...
+
+    @Test
+    public void testConvertStringOfThreeWords() {
+        User expected = new User("serega three zero",
+                "vany short boy",
+                "Van Clock Vandam",
+                "pushkin aleksandr lol");
+
+        User actual = stringToUserConverter.convert(
+                "serega three zero=vany short boy=Van Clock Vandam=pushkin aleksandr lol");
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java]: method(
argument
)

OR 

method(argument)

**Review by VladOsipov — CHANGES_REQUESTED** [review]

File: src/test/java/converter/StringToUserConverterTest.java [hunk 56]
```diff
@@ -0,0 +1,76 @@
...
+import static org.junit.Assert.assertNotNull;
+
+public class StringToUserConverterTest {
+    private final StringToUserConverter stringToUserConverter = new StringToUserConverter();
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testConvertEmptyString() {
+        stringToUserConverter.convert("");
+    }
+    @Test(expected = IllegalArgumentException.class)
```
Comment (VladOsipov) [inline src/test/java/converter/StringToUserConverterTest.java:16]: methods should be separated with empty line

**Review by VladOsipov — APPROVED** [review]
